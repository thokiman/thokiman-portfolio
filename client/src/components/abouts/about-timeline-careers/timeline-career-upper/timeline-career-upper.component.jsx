import React from "react";
import { GiCeremonialMask, GiGasMask, GiTotemMask } from "react-icons/gi";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import {
  toggleYear2016Click,
  toggleYear2018Click,
  toggleYearCurrentClick
} from "../../../../redux/about/about.actions";
import {
  selectIsYear2016Click,
  selectIsYear2018Click,
  selectIsYearCurrentClick
} from "../../../../redux/about/about.selectors";
import TimelineCareerLink from "../timeline-career-link/timeline-career-link.component";
import { UpperTimelineHeaderContainer } from "./timeline-career-upper.styles";


export const TimelineCareerUpper = ({
  isYear2016Click,
  isYear2018Click,
  isYearCurrentClick,
  toggleYear2016Click,
  toggleYear2018Click,
  toggleYearCurrentClick,
}) => {
  return (
    <UpperTimelineHeaderContainer>
      <TimelineCareerLink
        isClick={isYear2016Click}
        yearClick={toggleYear2016Click}
        TimelineIcon={GiCeremonialMask}
        placement="upper"
        year="2016"
      />
      <TimelineCareerLink
        isClick={isYear2018Click}
        yearClick={toggleYear2018Click}
        TimelineIcon={GiTotemMask}
        placement="upper"
        year="2018"
      />
      <TimelineCareerLink
        isClick={isYearCurrentClick}
        yearClick={toggleYearCurrentClick}
        placement="upper"
        TimelineIcon={GiGasMask}
        year="Today"
      />
    </UpperTimelineHeaderContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  isYear2016Click: selectIsYear2016Click,
  isYear2018Click: selectIsYear2018Click,
  isYearCurrentClick: selectIsYearCurrentClick,
});

const mapDispatchToProps = (dispatch) => ({
  toggleYear2016Click: () => dispatch(toggleYear2016Click()),
  toggleYear2018Click: () => dispatch(toggleYear2018Click()),
  toggleYearCurrentClick: () => dispatch(toggleYearCurrentClick()),
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TimelineCareerUpper);
