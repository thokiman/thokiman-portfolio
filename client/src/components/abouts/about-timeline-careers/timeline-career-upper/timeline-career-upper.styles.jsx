import styled from "styled-components";

export const UpperTimelineHeaderContainer = styled.div`
  grid-area: upper-timeline-header-container;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: 1fr;
  margin: 20px 0;
  cursor: pointer;
  height: 100%;
`;
UpperTimelineHeaderContainer.displayName = "UpperTimelineHeaderContainer";
