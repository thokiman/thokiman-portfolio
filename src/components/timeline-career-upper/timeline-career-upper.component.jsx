import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { GiCeremonialMask, GiTotemMask, GiGasMask } from "react-icons/gi";

import {
  selectIsYear2016Click,
  selectIsYear2018Click,
  selectIsYearCurrentClick,
} from "../../redux/about/about.selectors";
import TimelineCareerLink from "../timeline-career-link/timeline-career-link.component";
import "./timeline-career-upper.styles.scss";
import {
  toggleYear2016Click,
  toggleYear2018Click,
  toggleYearCurrentClick,
} from "../../redux/about/about.actions";

const TimelineCareerUpper = ({
  isYear2016Click,
  isYear2018Click,
  isYearCurrentClick,
  toggleYear2016Click,
  toggleYear2018Click,
  toggleYearCurrentClick,
}) => {
  return (
    <div className="upper-timeline-header-container">
      <TimelineCareerLink
        isClick={isYear2016Click}
        yearClick={toggleYear2016Click}
        TimelineIcon={GiCeremonialMask}
        placement="upper"
        year="2016"
      />
      <TimelineCareerLink
        isClick={isYear2018Click}
        yearClick={toggleYear2018Click}
        TimelineIcon={GiTotemMask}
        placement="upper"
        year="2018"
      />
      <TimelineCareerLink
        isClick={isYearCurrentClick}
        yearClick={toggleYearCurrentClick}
        placement="upper"
        TimelineIcon={GiGasMask}
        year="Today"
      />
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  isYear2016Click: selectIsYear2016Click,
  isYear2018Click: selectIsYear2018Click,
  isYearCurrentClick: selectIsYearCurrentClick,
});

const mapDispatchToProps = (dispatch) => ({
  toggleYear2016Click: () => dispatch(toggleYear2016Click()),
  toggleYear2018Click: () => dispatch(toggleYear2018Click()),
  toggleYearCurrentClick: () => dispatch(toggleYearCurrentClick()),
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TimelineCareerUpper);
