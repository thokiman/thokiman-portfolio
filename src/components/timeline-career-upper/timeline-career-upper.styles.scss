.upper-timeline-header-container {
    grid-area: upper-timeline-header-container;
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    grid-template-rows: 1fr;

    margin: 20px 0;
    cursor: pointer;
    height: 100%;

}