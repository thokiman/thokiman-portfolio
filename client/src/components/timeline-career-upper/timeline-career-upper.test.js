import React from "react";
import { shallow } from "enzyme";

import { TimelineCareerUpper } from "./timeline-career-upper.component";

describe("TimelineCareerUpper component", () => {
  let wrapper;
  let mockToggleYear2016Click;
  let mockToggleYear2018Click;
  let mockToggleYearCurrentClick;
  const mockFalseBool = false;
  const mockTrueBool = true;
  beforeEach(() => {
    mockToggleYear2016Click = jest.fn();
    mockToggleYear2018Click = jest.fn();
    mockToggleYearCurrentClick = jest.fn();
    const mockProps = {
      isYear2016Click: mockFalseBool,
      isYear2018Click: mockFalseBool,
      isYearCurrentClick: mockTrueBool,
      toggleYear2016Click: mockToggleYear2016Click,
      toggleYear2018Click: mockToggleYear2018Click,
      toggleYearCurrentClick: mockToggleYearCurrentClick,
    };
    wrapper = shallow(<TimelineCareerUpper {...mockProps} />);
  });

  it("should render TimelineCareerUpper component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
