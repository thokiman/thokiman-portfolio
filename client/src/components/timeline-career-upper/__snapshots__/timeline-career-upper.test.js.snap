// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimelineCareerUpper component should render TimelineCareerUpper component 1`] = `ShallowWrapper {}`;
